### PR TITLE
At parse time, make sure that offset-write values can fit into 32-bit integers

### PR DESCRIPTION
**Pull request by sil2100**
(LP: #1617443).

### CODE REVIEW

File: ubuntu_image/parser.py [hunk 1]
```diff
@@ -64,6 +65,16 @@ def __call__(self, v):
             ]
 
 
+def Size32bit(v):
```
Comment (warsaw) [inline ubuntu_image/parser.py]: What do you think about giving `as_size()` optional min and max parameters, e.g.

```
def as_size(size, min=0, max=None)
```

Then `Size32bit()` could be implemented easily as `return as_size(v, max=GiB(4))`

We could then test the corner cases of `as_size()` in test_helpers.py.  `max` would be exclusive, e.g. the range would be [min, max)

I also wonder if it wouldn't just be easier to check the range at the end of `as_size()` by doing

```
if not (min <= converted_value < max):
    raise ValueError
```

We probably wouldn't need the struct module then.  What do you think?

Comment (sil2100) [inline ubuntu_image/parser.py]: I can do it as proposed, I just thought that using the struct module is more elegant than doing value comparison. But of course 'elegant' is of course a rather relative term. I also saw that the struct module is anyway used in ubuntu-image, so didn't see any harm in using it here as well.

I can modify it per your proposition of course, especially if you have any future ideas of using the min and max parameters for something else.

File: ubuntu_image/helpers.py [hunk 2]
```diff
@@ -58,21 +58,26 @@ def straight_up_bytes(count):
...
+        if mo is None:
+            raise ValueError(size)
+        size_in_bytes = mo.group(1)
+        value = {
+            '': straight_up_bytes,
+            'G': GiB,
+            'M': MiB,
+            }[mo.group(2)](int(size_in_bytes))
+    if ((isinstance(min, int) and not value >= min) or
+            (isinstance(max, int) and not value < max)):
```
Comment (warsaw) [inline ubuntu_image/helpers.py]: Thanks!  I think we can always assume min will be an int.  We do have to check whether max is None, but if it's not, we can also assume it's an int.

File: ubuntu_image/helpers.py [hunk 3]
```diff
@@ -58,21 +58,26 @@ def straight_up_bytes(count):
...
+        raise ValueError
```
Comment (warsaw) [inline ubuntu_image/helpers.py]: I'd like to add a message to the exception.

File: ubuntu_image/tests/test_helpers.py [hunk 4]
```diff
@@ -76,6 +76,19 @@ def test_g(self):
     def test_bytes(self):
         self.assertEqual(as_size('801'), 801)
 
+    def test_size_min(self):
+        self.assertEqual(as_size(5, min=4), 5)
+        self.assertRaises(ValueError, as_size, 3, min=4)
```
Comment (warsaw) [inline ubuntu_image/tests/test_helpers.py]: I just want to add one test to prove that value==max will raise an exception.

**Comment by warsaw** [comment]
Superseded by #86 - Mostly your work with just a few tweaks and additional unit tests.  Thanks!
